### PR TITLE
Make dirname configurable for all datasets.

### PR DESCRIPTION
**Pull request by jacquerie**
### Summary

Some datasets use an intermediate folder in which to store the data after untarring/unzipping. The name of this folder was not configurable, unlike the other datasets, which prevented users from passing their own absolute paths unless they defined the appropriate symbolic links.

### Related Issues

Closes https://github.com/keras-team/keras/issues/11385.

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [x] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [ ] This PR is backwards compatible [y/n]
- [x] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR. We won't merge this PR because we don't think it would be relevant to sufficiently many users, and because API arguments should not be conditioned on implementation details.

**Comment by hyzhak** [comment]
ok, but what about path for datasets? Is there anyway to specify path destination in Keras?

**Comment by numericx** [comment]
> ok, but what about path for datasets? Is there anyway to specify path destination for datasets in Keras?

`import os
import gzip
import numpy as np
from keras.utils.data_utils import get_file

# DOC: https://github.com/keras-team/keras/blob/v2.12.0/keras/datasets/fashion_mnist.py#L28-L111
def load_data(path):
    """Loads the Fashion-MNIST dataset.

    This is a dataset of 60,000 28x28 grayscale images of 10 fashion categories,
    along with a test set of 10,000 images. This dataset can be used as
    a drop-in replacement for MNIST.

    The classes are:

    | Label | Description |
    |:-----:|-------------|
    |   0   | T-shirt/top |
    |   1   | Trouser     |
    |   2   | Pullover    |
    |   3   | Dress       |
    |   4   | Coat        |
    |   5   | Sandal      |
    |   6   | Shirt       |
    |   7   | Sneaker     |
    |   8   | Bag         |
    |   9   | Ankle boot  |

    Args:
      path: path where to cache the dataset locally
        (relative to `~/.keras/datasets`).

    Returns:
      Tuple of NumPy arrays: `(x_train, y_train), (x_test, y_test)`.

    **x_train**: uint8 NumPy array of grayscale image data with shapes
      `(60000, 28, 28)`, containing the training data.

    **y_train**: uint8 NumPy array of labels (integers in range 0-9)
      with shape `(60000,)` for the training data.

    **x_test**: uint8 NumPy array of grayscale image data with shapes
      (10000, 28, 28), containing the test data.

    **y_test**: uint8 NumPy array of labels (integers in range 0-9)
      with shape `(10000,)` for the test data.

    Example:

    ```python
    (x_train, y_train), (x_test, y_test) = fashion_mnist.load_data()
    assert x_train.shape == (60000, 28, 28)
    assert x_test.shape == (10000, 28, 28)
    assert y_train.shape == (60000,)
    assert y_test.shape == (10000,)
    ```

    License:
      The copyright for Fashion-MNIST is held by Zalando SE.
      Fashion-MNIST is licensed under the [MIT license](
      https://github.com/zalandoresearch/fashion-mnist/blob/master/LICENSE).

    """    
    dirname = os.path.join(path, "datasets", "fashion-mnist")
    base = "https://storage.googleapis.com/tensorflow/tf-keras-datasets/"
    files = [
        "train-labels-idx1-ubyte.gz",
        "train-images-idx3-ubyte.gz",
        "t10k-labels-idx1-ubyte.gz",
        "t10k-images-idx3-ubyte.gz",
        ]

    paths = []
    for fname in files:
        paths.append(get_file(fname=fname, origin=base + fname, cache_subdir=dirname))

    with gzip.open(filename=paths[0], mode="rb") as lbpath:
        y_train = np.frombuffer(buffer=lbpath.read(), dtype=np.uint8, offset=8)

    with gzip.open(filename=paths[1], mode="rb") as imgpath:
        x_train = np.frombuffer(buffer=imgpath.read(), dtype=np.uint8, offset=16).reshape(
            len(y_train), 28, 28
        )

    with gzip.open(filename=paths[2], mode="rb") as lbpath:
        y_test = np.frombuffer(buffer=lbpath.read(), dtype=np.uint8, offset=8)

    with gzip.open(filename=paths[3], mode="rb") as imgpath:
        x_test = np.frombuffer(buffer=imgpath.read(), dtype=np.uint8, offset=16).reshape(
            len(y_test), 28, 28
        )

    return (x_train, y_train), (x_test, y_test)`
